qa/tasks/cephfs/mount.py

```diff
@@ -182,8 +182,9 @@ def _create_mntpt(self):
                                timeout=60)
         # Use 0000 mode to prevent undesired modifications to the mountpoint on
         # the local file system.
-        self.client_remote.run(args=f'chmod 0000 {self.hostfs_mntpt}',
-                               timeout=60)
+        if not self.is_mounted():
+            self.client_remote.run(args=f'chmod 0000 {self.hostfs_mntpt}',
+                    timeout=60)
```

Comment:
I think using is_mounted() is not a good option because the method relies on a instance variable the value of which can change by accident. I have faced this issue before. IIRC, I wrote a patch/PR for this but did not post it, I'll check if I can find it.

Reply:
Found the commit, it was part of a branch for xfstests-dev.py for which I have not created a PR yet. The commit - 
https://github.com/rishabh-d-dave/ceph/commit/e31d5ffb3d047cfff1f9ee7d7daf7b0d38b7df2c

Reply:
@vshankar Should we include this commit/patch here? I think it's best to backport it, so a separate PR is suited.

Reply:
@rishabh-d-dave would recheck_mount/validate_mount sounds more specific?

Reply:
I don't remember the exact reason but if IIRC I did not wanted to delete `self.mounted` because it would need more changes (that is: modifying every code location where `self.mounted` is used) since I was at that time working on something else. 

Anyways, it makes sense to remove self.mounted entirely. I'll make that change.

Reply:
I'll put a separate PR since current changes on this PR would circumvent the chmod issue.